### PR TITLE
fix: update sysinfo and fix fd leak on linux

### PR DESCRIPTION
**Pull request by fraidev**
Update sysinfo to 0.31.4 and fix file descriptor leak on Linux. 
`sysinfo` leaks file descriptors on Linux if the limit is not set.

Spu Before:

```
▷ ls /proc/570230/fd/ | wc -l
1114
```

Spu After:
```
▷ ls /proc/574099/fd/ | wc -l
19
```

### CODE REVIEW

File: crates/fluvio-cluster/src/cli/diagnostics.rs [hunk 1]
```diff
@@ -452,13 +453,17 @@ impl ProcessInfo {
...
-                processes.push(ProcessInfo {
-                    pid: pid.as_u32(),
-                    name: process.name().to_string(),
-                    disk_usage: format!("{:?}", process.disk_usage()),
-                    cmd: format!("{:?}", process.cmd()),
-                });
+            let process_name = process.name().to_str();
+
+            if let Some(process_name) = process_name {
+                if process_name.contains("fluvio") {
```
Comment (sehz) [inline crates/fluvio-cluster/src/cli/diagnostics.rs:459]: can you make `fluvio` as constant?

File: crates/fluvio-cluster/src/delete.rs [hunk 2]
```diff
@@ -149,16 +150,23 @@ impl ClusterUninstaller {
         pb.set_message("Uninstalling fluvio local components");
 
         let kill_proc = |name: &str, command_args: Option<&[String]>| {
+            sysinfo::set_open_files_limit(0);
             let mut sys = System::new();
-            sys.refresh_processes(); // Only load what we need.
-            for process in sys.processes_by_exact_name(name) {
+            sys.refresh_processes(sysinfo::ProcessesToUpdate::All); // Only load what we need.
```
Comment (sehz) [inline crates/fluvio-cluster/src/delete.rs:155]: this seems like repeating. can you change into common util?

**Review by sehz — CHANGES_REQUESTED** [review]
LGTM.  Couple of minor nits

File: crates/fluvio-cluster/src/delete.rs (hunk 2 above)
Comment (fraidev) [reply crates/fluvio-cluster/src/delete.rs]: Sure! I created a `process` mod.

File: crates/fluvio-cluster/src/cli/diagnostics.rs (hunk 1 above)
Comment (fraidev) [reply crates/fluvio-cluster/src/cli/diagnostics.rs]: Done!

**Review by sehz — APPROVED** [review]

**Pull request merged**